lua/telescope/make_entry.lua

```diff
@@ -178,12 +178,18 @@ function make_entry.gen_from_buffer(opts)
       return nil
     end
 
+    if opts.shorten_path then
+      bufname = utils.path_shorten(bufname)
+    end
+
     return {
       valid = true,
 
       value = bufname,
-      ordinal = bufnr_str .. " : " .. bufname,
-      display = bufnr_str .. " : " .. bufname,
+      ordinal = string.format("%" .. opts.bufnr_width ..
+                              "s : %s", bufnr_str, bufname),
+      display = string.format("%" .. opts.bufnr_width ..
+                              "s : %s", bufnr_str, bufname),
 
       bufnr = entry,
       filename = bufname,
```

Comment:
With `shorten_path`, you change the filename, which makes the previewer unable to find the buffer.

After you fix that, I think I'll merge. Very cool! TY

Reply:
Ah makes sense. Should be fixed.